### PR TITLE
[IMPL] Tracers [Closes #68]

### LINKED ISSUE #68: [SUGGESTION] Tracers

### PR DESCRIPTION
**Pull request by Wolfsurge**
If this code could be improved, just say how it could be improved, instead of going 'haha this code is shit'

### CODE REVIEW

File: src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public static Setting<To> to = new Setting<>("To", To.BODY).setDescription("Where to draw the line to");
+
+    public Tracers() {
+        super("Tracers", Category.VISUAL, "Draws lines to entities in the world");
+    }
+
+    @Override
+    public void onRender3D() {
+        // Draw lines to entities if they are valid
+        for(Entity entity : mc.world.loadedEntityList) {
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java:37]: You could use a forEach loop here

File: src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            if(isEntityValid(entity)) {
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java:38]: Add space after if

File: src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        }
+    }
+
+    /**
+     * Checks if an entity is valid
+     * @param entity The entity to check
+     * @return If the entity is valid
+     */
+    public boolean isEntityValid(Entity entity) {
+        return entity instanceof EntityOtherPlayerMP && players.getValue() || entity instanceof EntityLiving && !(entity instanceof EntityMob) && passive.getValue() || entity instanceof EntityMob && mobs.getValue();
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java:57]: Use EntityUtil here

File: src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    /**
+     * Gets where to draw the line to
+     * @param entityIn The entity
+     * @return The place to draw the line to
+     */
+    public float getHeight(Entity entityIn) {
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java:65]: No need to create methods if they're only going to be used once, could just move this check to the render event

File: src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            default:
+                return 0;
+        }
+    }
+
+    /**
+     * Place to draw the line to
+     */
+    enum To {
+        // Draw line to the entity's feet
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java:80]: Enums still follow java doc conventions (https://stackoverflow.com/questions/6549821/how-to-javadoc-a-classs-individual-enums)

File: src/main/java/cope/cosmos/util/string/ColorUtil.java [hunk 6]
```diff
@@ -54,4 +55,12 @@ public static Color rainbow(long offset, int alpha) {
 		int blue = rgb & 255;
 		return new Color(red, green, blue, alpha);
 	}
+
+	/**
+	 * Sets the color
+	 * @param color The color to set
+	 */
+	public static void glColor(Color color) {
```
Comment (linustouchtips) [inline src/main/java/cope/cosmos/util/string/ColorUtil.java:63]: Remove this method, just call the glMethod in it's place

**Comment by master7720** [comment]
make sure theses tracers have a strict bypass.

File: src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java (hunk 5 above)
Comment (Sxmurai) [reply src/main/java/cope/cosmos/client/features/modules/visual/Tracers.java:80]: enum visibility should also be set to public for consistency

**Pull request merged**